File: app/src/main/java/id/ac/uasbungaa/data/local/UserHelper.java
package id.ac.uasbungaa.data.local;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import java.sql.SQLException;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public final class UserHelper {
    private DatabaseHelper dbHelper;
    private SQLiteDatabase db;
    private static final String DATABASE_TABLE = "USER_200020016";

    public final void open() {
        this.db = this.dbHelper.getWritableDatabase();
    }

    public final void close() {
        this.dbHelper.close();
        SQLiteDatabase db = this.db;

        if (db.isOpen()) {
            db = this.db;
            db.close();
        }
    }

    @NotNull
    public final Cursor getAllUser() {
        SQLiteDatabase db = this.db;

        return db.query(DATABASE_TABLE, (String[]) null, (String) null, (String[]) null, (String) null, (String) null, "_id ASC");
    }

    public final long insert(@NotNull ContentValues values) {
        SQLiteDatabase db = this.db;

        return db.insert(DATABASE_TABLE, (String) null, values);
    }

    public final int update(@NotNull String id, @Nullable ContentValues values) {
        SQLiteDatabase db = this.db;

        return db.update(DATABASE_TABLE, values, "_id = ?", new String[]{id});
    }

    public final int delete(@NotNull String id) {
        SQLiteDatabase db = this.db;

        return db.delete(DATABASE_TABLE, "_id = ?", new String[]{id});
    }

    @Nullable
    public final Cursor login(@NotNull String username, @NotNull String password) {
        this.db = this.dbHelper.getReadableDatabase();
        SQLiteDatabase db = this.db;

        return db.query(DATABASE_TABLE, (String[]) null, "username = ? AND password = ?", new String[]{username, password}, (String) null, (String) null, (String) null);
    }

    public UserHelper(@NotNull Context context) {
        this.dbHelper = new DatabaseHelper(context);
    }
}

File: app/src/main/java/id/ac/uasbunga/data/local/UserHelper.kt
package id.ac.uasbunga.data.local

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import id.ac.uasbunga.data.local.DatabaseContract.USER_200020016.Companion.PASSWORD
import id.ac.uasbunga.data.local.DatabaseContract.USER_200020016.Companion.TABLE_NAME
import id.ac.uasbunga.data.local.DatabaseContract.USER_200020016.Companion.USERNAME
import id.ac.uasbunga.data.local.DatabaseContract.USER_200020016.Companion._ID
import java.sql.SQLException

class UserHelper(context: Context) {
    private var dbHelper: DatabaseHelper = DatabaseHelper(context)
    private lateinit var db: SQLiteDatabase

    companion object {
        private const val DATABASE_TABLE = TABLE_NAME
        private var INSTANCE: UserHelper? = null
        fun getInstance(context: Context): UserHelper =
            INSTANCE ?: synchronized(this) {
                INSTANCE ?: UserHelper(context)
            }
    }

    @Throws(SQLException::class)
    fun open() {
        db = dbHelper.writableDatabase
    }

    fun close() {
        dbHelper.close()
        if (db.isOpen) {
            db.close()
        }
    }

    fun getAllUser(): Cursor {
        return db.query(
            DATABASE_TABLE,
            null,
            null,
            null,
            null,
            null,
            "$_ID ASC"
        )
    }

    fun getUser(id: String): Cursor {
        return db.query(
            DATABASE_TABLE,
            null,
            "$_ID = ?",
            arrayOf(id),
            null,
            null,
            null,
        )
    }

    fun insert(values: ContentValues): Long {
        return db.insert(DATABASE_TABLE, null, values)
    }

    fun update(id: String, values: ContentValues?): Int {
        return db.update(DATABASE_TABLE, values, "$_ID = ?", arrayOf(id))
    }

    fun delete(id: String): Int {
        return db.delete(DATABASE_TABLE, "$_ID = ?", arrayOf(id))
    }

    fun login(username: String, password: String): Cursor? {
        db = dbHelper.readableDatabase
        return db.query(
            DATABASE_TABLE,
            null,
            "$USERNAME = ? AND $PASSWORD = ?",
            arrayOf(username, password),
            null,
            null,
            null
        )
    }
}